activesupport/lib/active_support/core_ext/module/delegation.rb

```diff
@@ -215,4 +214,64 @@ def delegate(*methods)
       module_eval(method_def, file, line)
     end
   end
+
+  # When building decorators, a common pattern may emerge:
+  #
+  #   class Partition
+  #     def initialize(first_event)
+  #       @events = [ first_event ]
+  #     end
+  #
+  #     def people
+  #       if @events.first.detail.people.any?
+  #         @events.collect { |e| Array(e.detail.people) }.flatten.uniq
+  #       else
+  #         @events.collect(&:creator).uniq
+  #       end
+  #     end
+  #
+  #     private
+  #       def respond_to_missing?(name, include_private = false)
+  #         @events.respond_to?(name, include_private)
+  #       end
+  #
+  #       def method_missing(method, *args, &block)
+  #         @events.send(method, *args, &block)
+  #       end
+  #   end
+  #
+  # With `Module#delegate_missing_to`, the above is condensed to:
+  #
+  #   class Partition
+  #     delegate_missing_to :@events
+  #
+  #     def initialize(first_event)
+  #       @events = [ first_event ]
+  #     end
+  #
+  #     def people
+  #       if @events.first.detail.people.any?
+  #         @events.collect { |e| Array(e.detail.people) }.flatten.uniq
+  #       else
+  #         @events.collect(&:creator).uniq
+  #       end
+  #     end
+  #   end
+  #
+  # The target can be anything callable withing the object. E.g. instance
+  # variables, methods, constants ant the likes.
+  def delegate_missing_to(target)
+    target = target.to_s
+    target = "self.#{target}" if DELEGATION_RESERVED_METHOD_NAMES.include?(target)
+
+    module_eval <<-RUBY, __FILE__, __LINE__ + 1
+      def respond_to_missing?(name, include_private = false)
+        #{target}.respond_to?(name, include_private)
+      end
+
+      def method_missing(method, *args, &block)
+        #{target}.send(method, *args, &block)
```

Comment:
@rafaelfranca that has some pretty questionable results -- e.g., external calls to the private Kernel methods will now work on this object. I guess you want local methods to be able to call private instance methods... but IMO, it seems fair to force those to be more explicit: we're composing around the object's public API, not injecting methods inside it. Arguably, it'd be weirder if we *could* call private methods, yet have a separate set of ivars.